### PR TITLE
Misc updates to `runtime_env` doc

### PR DESCRIPTION
**Pull request by zhe-thoughts**
1. `runtime_env` should be GA in 1.6 release?
2. Based on Ray client doc, `ray.init(...)` is the way to use Ray client

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [x] This PR is not tested :(

### CODE REVIEW

File: doc/source/advanced.rst [hunk 1]
```diff
@@ -427,11 +427,6 @@ To get information about the current available resource capacity of your cluster
 
 Runtime Environments
 -----------------------------------
-
-.. note::
-
-    This API is in beta and may change before becoming stable.
-
```
Comment (AmeerHajAli) [inline doc/source/advanced.rst:434]: I am not sure if it is non experimental anymore.

Comment (zhe-thoughts) [reply doc/source/advanced.rst:434]: We did announce it as GA in 1.6 release

Comment (architkulkarni) [reply doc/source/advanced.rst:434]: I think here we really wanted to just use the beta stability annotation, but there's no way to do annotate a single kwarg (in this case, `runtime_env`.). However, if the API is no longer going to change, then we can remove this.  My only concern would be the new pluggability changes, but even for those we aren't proposing changing the API for any existing options like `pip`, `conda`, etc.  @edoakes what do you think, are we OK to remove this?

**Review by architkulkarni — APPROVED** [review]

File: doc/source/advanced.rst (hunk 1 above)
Comment (zhe-thoughts) [reply doc/source/advanced.rst:434]: Chatted with @edoakes offline. I'll close the PR for now and we can remove this block after we stabilize the API more

**Pull request closed without merging**